src/main/scala/codacy/docker/api/utils/Timeout.scala

```diff
@@ -0,0 +1,22 @@
+package codacy.docker.api.utils
+
+import java.util.{Timer, TimerTask}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Future, Promise}
+import scala.util.Try
+
+trait Timeout {
+
+  def initTimeout[T](delay: Duration)(onTimeout: => T): Future[T] = {
```

Comment:
"initTimeout" sounds weird to me. I'd rather have it named "onTimeout", like you did for the parameter.